pkg/reconciler/reconciler.go

```diff
@@ -362,8 +363,17 @@ func WithPostHook(h hook.PostHook) Option {
 	}
 }
 
+// WithValueTranslator is an Option that configures a function that translates a
+// custom resource to the values passed to Helm.
+func WithValueTranslator(t values.Translator) Option {
```

Comment:
+1, reading this PR I don't really understand why a custom translator is required in addition to the mapper. Things that come to mind:
- access to metadata (name, ...) and status fields
- ability to use Go types by converting the `*unstructured.Unstructured` into the Go type for a CR.

However, having translator + mapper seems confusing. Perspectively, I think it would make more sense to introduce translator as the first-class citizen, and drop the `valueMapper` field. `WithValueMapper` could then return a function that sets the `valueTranslator` accordingly (relying on `FromUnstructured`).

Or am I completely off, and the difference is actually whether it gets executed pre/post the application of overrides?

Reply:
@misberner the rationale is in the PR description:
> This lets the user inject code which produces helm values based on the fetched object itself (unlike Mapper which can only see Values).
> This way the custom code can convert the object to a typed one and work with proper structs rather than a tree of maps to interface{}.

Basically it is so that we can have write type-safe code at least to read the CRs.